modules/client/src/connect.ts

```diff
@@ -0,0 +1,243 @@
+  IConnextClient,
+} from "./types";
+
+const exists = (obj: any): boolean => {
+  return !!obj && !!Object.keys(obj).length;
+};
+
+const createMessagingService = async (
+  messagingUrl: string,
+  logLevel: number,
+): Promise<IMessagingService> => {
+  // create a messaging service client
+  const messagingFactory = new MessagingServiceFactory({ logLevel, messagingUrl });
+  const messaging = messagingFactory.createService("messaging");
+  await messaging.connect();
+  return messaging;
+};
+
+const setupMultisigAddress = async (
+  node: NodeApiClient,
+  channelProvider: ChannelProvider,
+  log: Logger,
+): Promise<ChannelProvider> => {
+  const myChannel = await node.getChannel();
+
+  let multisigAddress: string;
+  if (!myChannel) {
+    log.debug("no channel detected, creating channel..");
+    const creationEventData: CFCoreTypes.CreateChannelResult = await new Promise(
+      async (res: any, rej: any): Promise<any> => {
+        const timer = setTimeout(
+          (): void => rej("Create channel event not fired within 30s"),
+          30000,
+        );
+        channelProvider.once(
+          CFCoreTypes.EventNames.CREATE_CHANNEL_EVENT as CFCoreTypes.EventName,
+          (data: CreateChannelMessage): void => {
+            clearTimeout(timer);
+            res(data.data);
+          },
+        );
+
+        const creationData = await node.createChannel();
+        log.debug(`created channel, transaction: ${stringify(creationData)}`);
+      },
+    );
+    multisigAddress = creationEventData.multisigAddress;
+  } else {
+    multisigAddress = myChannel.multisigAddress;
+  }
+  log.debug(`multisigAddress: ${multisigAddress}`);
+
+  channelProvider.multisigAddress = multisigAddress;
+  return channelProvider;
+};
+
+export const connect = async (opts: ClientOptions): Promise<IConnextClient> => {
+  const {
+    logLevel,
+    ethProviderUrl,
+    nodeUrl,
+    store,
+    mnemonic,
+    channelProvider: providedChannelProvider,
+  } = opts;
+  let { xpub, keyGen } = opts;
+
+  const log = new Logger("ConnextConnect", logLevel);
+
+  // setup ethProvider + network information
+  log.debug(`Creating ethereum provider - ethProviderUrl: ${ethProviderUrl}`);
+  const ethProvider = new providers.JsonRpcProvider(ethProviderUrl);
+  const network = await ethProvider.getNetwork();
+
+  // special case for ganache
+  if (network.chainId === 4447) {
+    network.name = "ganache";
+    // Enforce using provided signer, not via RPC
+    ethProvider.getSigner = (addressOrIndex?: string | number): any => {
+      throw { code: "UNSUPPORTED_OPERATION" };
+    };
+  }
+
+  // setup messaging and node api
+  let messaging: IMessagingService;
+  let node: NodeApiClient;
+  let config: GetConfigResponse;
+
+  // setup channelProvider
+  let channelProvider: ChannelProvider;
+
+  if (providedChannelProvider) {
+    channelProvider = providedChannelProvider;
+    if (!exists(channelProvider.config)) {
+      await channelProvider.enable();
```

Comment:
This could be replace by throwing an error as well but it wouldn't be backwards-compatible however channelProvider has only been implemented in the daicard which this change is included in the PR (https://github.com/ConnextProject/indra/pull/638/files#diff-e5c0bb91d250a5754dc0e76b4c9fa552R279)